extensions/wikia/Email/Controller/WatchedPageController.class.php

```diff
@@ -26,8 +33,11 @@ public function initEmail() {
 
 		$this->title = \Title::newFromText( $titleText, $titleNamespace );
 		$this->summary = $this->getVal( 'summary' );
-		$this->currentRevId = $this->getVal( 'currentRevId' );
-		$this->previousRevId = $this->getVal( 'previousRevId' );
+
+		if ( $this->title instanceof \Title ) {
+			$this->currentRevId = $this->title->getLatestRevID( \Title::GAID_FOR_UPDATE );
+			$this->previousRevId = $this->title->getPreviousRevisionID( $this->currentRevId, \Title::GAID_FOR_UPDATE );
+		}
```

Comment:
We don't have to handle edits separately, just test that we got a valid ```Title``` object from the 'title' parameter and make the lookups the default values for ```currentRevId``` and ```previousRevId```:
```php
$this->title = \Title::newFromText( $titleText, $titleNamespace );
$this->assertValidTitle();

$this->currentRevId = $this->getVal(
	'currentRevId',
	$this->title->getLatestRevID( \Title::GAID_FOR_UPDATE )
);
$this->previousRevId = $this->getVal(
	'previousRevId',
	$this->title->getPreviousRevisionID( $this->currentRevId, \Title::GAID_FOR_UPDATE )
);
```
It doesn't happen a ton (about 20 times last year on glee, query to follow) but the changes above are pretty straightforward.  Also, it means we don't query the master DB when we don't need to (which is most of the time since most changes are edits).

Here's the nasty query I used to look at edit clumping per 5s interval:
```sql
SELECT
	rev_page, 
	from_unixtime(unix_timestamp(str_to_date(rev_timestamp, '%Y%m%d%H%i%s'))
	 - (unix_timestamp(str_to_date(rev_timestamp, '%Y%m%d%H%i%s')) % 5)) as ts,
	count(*) AS cnt
FROM revision, page
WHERE rev_page=page_id
	AND page_namespace = 0
GROUP BY rev_page, ts
HAVING cnt > 1
ORDER BY ts DESC
LIMIT 40;
```